WayPoint left_boundary
WayPoint right_boundary
int32    direction

